### PR TITLE
Make TPM key templates and key creation function accessible

### PR DESCRIPTION
**Pull request by shjala**
This change makes the TPM key templates and key creation function accessible to other package, for example vtpm or recovertpm to reduce code duplication. There is no functional changes in this PR.

### CODE REVIEW

File: pkg/pillar/cmd/tpmmgr/tpmmgr.go [hunk 1]
```diff
@@ -77,12 +77,15 @@ var (
...
 	quoteKeyFile = types.CertificateDirname + "/attest.key.pem"
 
-	// this PCRSelections is used as an entropy to generate keys and the selection
+	// PCRSelection is used as an entropy to generate keys and the selection
 	// of PCRs do not matter as well as the contents but PCR[7] is not changed often
 	// on our devices
-	pcrSelection     = tpm2.PCRSelection{Hash: tpm2.AlgSHA256, PCRs: []int{7}}
-	pcrListForQuote  = tpm2.PCRSelection{Hash: tpm2.AlgSHA256, PCRs: []int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15}}
-	defaultKeyParams = tpm2.Public{
+	PcrSelection = tpm2.PCRSelection{Hash: tpm2.AlgSHA256, PCRs: []int{7}}
```
Comment (milan-zededa) [inline pkg/pillar/cmd/tpmmgr/tpmmgr.go:83]: Do you plan to import this from somewhere? We usually don't import `../cmd/..` packages as they are intended to build the final "binaries". Would it make sense to move the exported variables and functions to `pillar/evetpm`?

Comment (shjala) [reply pkg/pillar/cmd/tpmmgr/tpmmgr.go:83]: good point, yes!

File: pkg/pillar/evetpm/tpm.go [hunk 2]
```diff
@@ -119,6 +119,109 @@ var (
 	// measurefsTpmEventLog is the file containing the event log from the measure-config.
 	// it is not a constant due to test usage.
 	measurefsTpmEventLog = types.PersistStatusDir + "/measurefs_tpm_event_log"
+
+	// PCRSelection is used as an entropy to generate keys and the selection
```
Comment (eriknordmark) [inline pkg/pillar/evetpm/tpm.go:123]: Name in comment doesn't match name of variable.

**Comment by eriknordmark** [comment]
> There is functional changes in this PR.

Did you mean that there are NO functional changes? If not, please specify the functional changes.

**Comment by shjala** [comment]
> > There is functional changes in this PR.
> 
> Did you mean that there are NO functional changes? If not, please specify the functional changes.

sorry, there is no functional changes.

File: pkg/pillar/evetpm/tpm.go (hunk 2 above)
Comment (shjala) [reply pkg/pillar/evetpm/tpm.go]: fixed.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**